x/twap/logic_test.go

```diff
@@ -1044,9 +1043,85 @@ func (s *TestSuite) TestUpdateRecords() {
 				},
 			},
 		},
-		// TODO: complete multi-asset pool tests:
-		// "multi-asset pool; pre-set at t and t + 1; creates new records": {},
-		// "multi-asset pool; pre-set at t and t + 1; pre-existing records some with error and some with too large spot price, overwrites erorr time":                        {},
+		"multi-asset pool; pre-set at t and t + 1; creates new records": {
+			preSetRecords: []types.TwapRecord{threeAssetRecordAB, tPlus10sp5ThreeAssetRecordAB},
```

Comment:
For multi-asset pools, there should be at least 3 records at a given time.

Assume a 3 asset pool with assets A, B and C.

Then, if swap occurs at time t, we should have AB, AC, and BC records (for each pair).

Given that this test is meant for time t and t + 1, we need 6 records.

We should also have 3 spot price overrides and 3 most recent records.

Reply:
Ax thank you! Already updated could u pls check